internal/playwright/config.go

```diff
@@ -225,6 +227,17 @@ func shardInSuites(rootDir string, suites []Suite, ccy int) ([]Suite, error) {
 
 		testFiles := fpath.ExcludeFiles(files, excludedFiles)
 
+		if s.ShardGrepEnabled && (s.Params.Grep != "" || s.Params.GrepInvert != "") {
+			var unmatched []string
+			testFiles, unmatched = grep.MatchFiles(os.DirFS(rootDir), testFiles, s.Params.Grep, s.Params.GrepInvert)
+			if len(testFiles) == 0 {
+				log.Warn().Str("suiteName", s.Name).Str("grep", s.Params.Grep).Msg("No files match the configured grep expressions")
```

Comment:
I think we need to error. I know the cypress sharding feature doesn't error either, but this is what happens otherwise:
```
❯ saucectl run
15:32:44 WRN No files match the configured grep expressions grep=@fast suiteName="Firefox Win"
15:32:44 INF Running Playwright in Sauce Labs
15:32:44 INF Archive created. durationMs=103 fileCount=70 longestPathLength=185 size=2521830
15:32:44 INF Checking if /var/folders/ck/mccvpn5x4lz6strcwjjmb5000000gn/T/saucectl-app-payload-502245848/app.zip has already been uploaded previously
15:32:44 INF Checksum: 7dc7736f774493c05e7c2495a05efb8a6714eb58510f2d40e28aac454b64a1eb
15:32:46 INF Project uploaded. durationMs=1423 storageId=59565837-ccb1-49f5-a5e8-ee50c62afec9
15:32:46 INF Checking if /var/folders/ck/mccvpn5x4lz6strcwjjmb5000000gn/T/saucectl-app-payload-502245848/config.zip has already been uploaded previously
15:32:46 INF Checksum: 559fe591329206809093957c721a200cfaf03586c1db9a1809878e35c8c0ae35
15:32:46 INF Runner Config uploaded. durationMs=571 storageId=a34f628d-ca0e-46e6-a14d-8cdcaec61bc4
15:32:46 INF Launching workers. concurrency=10


  Results:
```

We perform a lot of the setup (e.g. archiving) just to fail at a later point. It'd be much better to error out right away (and cypress as well).

Reply:
I think this point is still valid. If you do it [this way](https://github.com/saucelabs/saucectl/pull/800/files#diff-a2f8cf61026ba21eee0fcfa6cdb43be4deef7d5629b41005c14a4660915c979dR328), and you have multiple suites defined, some with and without sharding, the ones that have sharding incorrectly set-up will be skipped and the correctly set up suites will run. So you still have to abort your run, because it's misconfigured. It'd be better to truly fail here and provide feedback to the user as soon as possible.

Reply:
Correct ! That's fixed !